service/service.go

```diff
@@ -118,102 +111,103 @@ func (svc *Service) HandleMessage(ctx context.Context, message kafka.Message) (s
-			DatasetAPIAuthToken:   svc.DatasetAPIAuthToken,
 			InstanceID:            instanceID,
 			Line:                  line,
-			MaxRetries:            svc.MaxRetries,
 			TimeColumn:            timeColumn,
 			CodelistMap:           codelistMap,
 		}
 
 		lineDimensions, err := dim.Extract()
 		if err != nil {
-			log.ErrorC("encountered error retrieving dimensions", err, log.Data{"instance_id": instanceID, "csv_line": line})
+			log.Event(ctx, "encountered error retrieving dimensions", log.ERROR, log.Error(err), log.Data{"instance_id": instanceID, "csv_line": line})
 			return instanceID, err
 		}
 
-		for _, request := range lineDimensions {
-			if err := request.Post(ctx, svc.HTTPClient); err != nil {
-				log.ErrorC("encountered error sending request to datset api", err, log.Data{"instance_id": instanceID, "csv_line": line})
+		for _, optionToPost := range lineDimensions {
+			if err := svc.DatasetClient.PostInstanceDimensions(ctx, svc.AuthToken, instanceID, optionToPost); err != nil {
+				log.Event(ctx, "encountered error sending request to datset api", log.ERROR, log.Error(err), log.Data{"instance_id": instanceID, "csv_line": line})
 				return instanceID, err
 			}
 		}
 
 		numberOfObservations++
 	}
-
-	log.Trace("a count of the number of observations", log.Data{"instance_id": instanceID, "number_of_observations": numberOfObservations})
-
-	jobInstance := instance.NewJobInstance(svc.AuthToken, svc.DatasetAPIURL, svc.DatasetAPIAuthToken, instanceID, numberOfObservations, headerRow, svc.MaxRetries)
-
-	// PUT request to dataset API to pass the header row and the
-	// number of observations that exist against this job instance
-	if err := jobInstance.PutData(ctx, svc.HTTPClient); err != nil {
-		log.ErrorC("encountered error sending request to the dataset api", err, log.Data{"instance_id": instanceID, "number_of_observations": numberOfObservations})
+	log.Event(ctx, "a count of the number of observations", log.INFO, log.Data{"instance_id": instanceID, "number_of_observations": numberOfObservations})
+
+	// PUT request to dataset API to pass the header row and the number of observations that exist against this job instance
+	err = svc.DatasetClient.PutInstanceData(
+		ctx,
+		svc.AuthToken,
+		instanceID,
+		dataset.JobInstance{
+			HeaderNames:          headerRow,
+			NumberOfObservations: numberOfObservations,
+		})
+	if err != nil {
+		log.Event(ctx, "encountered error sending request to the dataset api", log.ERROR, log.Error(err), log.Data{"instance_id": instanceID, "number_of_observations": numberOfObservations})
 		return instanceID, err
 	}
+	log.Event(ctx, "successfully sent request to dataset API", log.INFO, log.Data{"instance_id": instanceID, "number_of_observations": numberOfObservations})
 
-	log.Trace("a list of headers", log.Data{"instance_id": instanceID, "header_row": headerRow})
+	log.Event(ctx, "a list of headers", log.INFO, log.Data{"instance_id": instanceID, "header_row": headerRow})
```

Comment:
Can get rid of this log - it is duplicated on line 102 and does not have much value anyway